src/Hng.Web/Repo/Interface/IGenericRepository.cs

```diff
@@ -0,0 +1,17 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Hng.Web.Repo.Interface
+{
+    public interface IGenericRepository<T> where T : class
```

Comment:
@davideshett You should have a base class called `EntityBase` that has a single `Guid Id` property. All other entities must then inherit this base class. So that you can now have `IGenericRepository<T> where T : EntityBase`.

That is the right approach. With this I can get `IGenericRepository<String>` which makes no sense.

Reply:
This part is a bit confusing boss...i"m stuck here a bit

Reply:
You don't understand inheritance?